config/initializers/sidekiq.rb

```diff
@@ -1,7 +1,7 @@
 if $0.include?('sidekiq')
   Sidekiq.configure_server do |config|
     config.redis = {
-      url: "redis://#{Settings.redis_url}",
+      url: ENV['REDIS_URL'] || "redis://#{Settings.redis_url}",
```

Comment:
I don't think this is necessary. The env variable `SETTINGS__REDIS_URL` will override what is in the settings config file. You can try it like this:

```
docker-compose run -e SETTINGS__REDIS_URL=foo rails rails c
[1] pry(main)> Settings.redis_url
=> "foo"
```